hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpcengine/nio/NioAsyncSocketBuilder.java

```diff
@@ -73,10 +81,34 @@ public <T> boolean setIfSupported(Option<T> option, T value) {
         return options.setIfSupported(option, value);
     }
 
-    public NioAsyncSocketBuilder setReceiveBufferIsDirect(boolean receiveBufferIsDirect) {
+    @Override
+    public AsyncSocketBuilder setSSLEngineFactory(SSLEngineFactory sslEngineFactory) {
```

Comment:
We need to ensure here that we are running in EE: this can be done by checking if the constructor to the TlsNioAsyncSocket could be loaded.